src/messageprocessor.ts

```diff
@@ -47,20 +61,19 @@ export class MessageProcessor {
         const result = new MessageProcessorMatrixResult();
 
         let content = msg.content;
-        // embeds are markdown formatted, thus inserted before
-        // for both plaintext and markdown
-        content = this.InsertEmbeds(content, msg);
         
         // for the formatted body we need to parse markdown first
         // as else it'll HTML escape the result of the discord syntax
-        let contentPostmark = marked(content);
+        let contentPostmark = marked(content).replace(/\n/g, "<br>").replace(/(<br>)?<\/p>(<br>)?/g, "</p>");
         
         // parse the plain text stuff
+        content = this.InsertEmbeds(content, msg);
```

Comment:
Why did we move InsertEmbeds? The comment above suggests we really don't want to be doing that because otherwise we are inserting embeds after generating the HTML. (Though I must admit the tests didn't break, which is concerning)